packager/mpd/base/xml/xml_node.cc

```diff
@@ -540,16 +544,18 @@ bool RepresentationXmlNode::AddAudioChannelInfo(const AudioInfo& audio_info) {
       // Convert EC3 channel map into string of hexadecimal digits. Spec: DASH-IF
       // Interoperability Points v3.0 9.2.1.2.
       const uint16_t ec3_channel_map =
-        base::HostToNet16(codec_data.channel_mask());
-      audio_channel_config_value =
-        base::HexEncode(&ec3_channel_map, sizeof(ec3_channel_map));
+          absl::big_endian::FromHost16(codec_data.channel_mask());
+      uint8_t byte1 = ec3_channel_map & 0xFF;
+      uint8_t byte2 = ec3_channel_map >> 8;
+      audio_channel_config_value = absl::StrFormat("%02X%02X", byte1, byte2);
```

Comment:
couldn't find a better way to get the `base::HexEncode` here in big endian format, is this the best way to proceed?

Reply:
Maybe add another helper to packager/utils/hex_parser.  I'm not certain you need to decompose it into individual bytes if you've already done a swap with absl.  Just FromHost16 and StrFormat("%04X") should be fine, right?  Or FromHost16 and BytesToHexString?

Reply:
That's where I started and that didn't work, the byte order was reversed, but looks like I was overthinking it. `StrFormat("%04X", ...)` works by itself, without `FromHost16`, going to big endian order is only important if we go byte at a time like the manual formatting.